manifests/topic.pp

```diff
@@ -7,21 +7,28 @@
 # This defined type is used to manage the creation of kafka topics.
 #
 define kafka::topic(
-  $ensure             = '',
-  $zookeeper          = '',
-  $replication_factor = 1,
-  $partitions         = 1,
-  $bin_dir            = '/opt/kafka/bin',
+  String                        $ensure             = '',
+  String                        $zookeeper          = '',
+  Integer                       $replication_factor = 1,
+  Integer                       $partitions         = 1,
+  String                        $bin_dir            = '/opt/kafka/bin',
+  Optional[Hash[String,String]] $config             = undef,
 ) {
 
   $_zookeeper          = "--zookeeper ${zookeeper}"
   $_replication_factor = "--replication-factor ${replication_factor}"
   $_partitions         = "--partitions ${partitions}"
 
+  if $config {
+    $_config = inline_template('<% @config.each do |key, value|%> --config <%= key %>=<%= value %><% end %>')
```

Comment:
it should be possible to achieve the same with the `join()` function, right?

Reply:
Hm ... yes. Will change it